xm-commons-topic/src/main/java/com/icthh/xm/commons/topic/service/TopicConfigurationService.java

```diff
@@ -0,0 +1,109 @@
+package com.icthh.xm.commons.topic.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.icthh.xm.commons.config.client.api.RefreshableConfiguration;
+import com.icthh.xm.commons.topic.domain.ConsumerHolder;
+import com.icthh.xm.commons.topic.domain.TopicConfig;
+import com.icthh.xm.commons.topic.domain.TopicConsumersSpec;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.util.AntPathMatcher;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+@Component
+public class TopicConfigurationService implements RefreshableConfiguration {
+
+    private static final String TENANT_NAME = "tenant";
+
+    private AntPathMatcher matcher = new AntPathMatcher();
+    private ObjectMapper ymlMapper = new ObjectMapper(new YAMLFactory());
+
+    private final String configPath;
+    private final TopicManagerService topicManagerService;
+    private final DynamicConsumerConfigurationService dynamicConsumerConfigurationService;
+
+    @Getter
+    private Map<String, Map<String, ConsumerHolder>> tenantTopicConsumers = new ConcurrentHashMap<>();
+
+    public TopicConfigurationService(@Value("${spring.application.name}") String appName,
+                                     TopicManagerService topicManagerService,
+                                     DynamicConsumerConfigurationService dynamicConsumerConfigurationService) {
+        this.configPath = "/config/tenants/{tenant}/" + appName + "/topic-consumers.yml";
+        this.topicManagerService = topicManagerService;
+        this.dynamicConsumerConfigurationService = dynamicConsumerConfigurationService;
+    }
+
+    @Override
+    public void onRefresh(String updatedKey, String config) {
+        refreshConfig(updatedKey, config);
+    }
+
+    @Override
+    public boolean isListeningConfiguration(String updatedKey) {
+        return matcher.match(configPath, updatedKey);
+    }
+
+    @Override
+    public void onInit(String configKey, String configValue) {
+        if (isListeningConfiguration(configKey)) {
+            refreshConfig(configKey, configValue);
+        }
+    }
+
+    private void refreshConfig(String updatedKey, String config) {
+        String tenantKey = extractTenant(updatedKey);
+        Map<String, ConsumerHolder> existingConsumers = getTenantConsumers(tenantKey);
+
+        if (StringUtils.isEmpty(config)) {
+            topicManagerService.stopAllTenantConsumers(tenantKey, existingConsumers);
```

Comment:
we do not need manual actions, maybe better just implement DynamicConsumerConfiguration?

Reply:
I mean TopicConfigurationService must implements DynamicConsumerConfiguration and just call refresh consumers

Reply:
done